### PR TITLE
Replace embedded telem implementation with mapbox-mobile-events library

### PR DESCRIPTION
**Pull request by boundsj**
Depends on https://github.com/mapbox/mapbox-events-ios/pull/27

### CODE REVIEW

**Comment by akitchen** [comment]
Does this PR need to be updated to point to a different submodule SHA? I tried checking it out, and `git submodule update --init` returned:

```
error: Server does not allow request for unadvertised object 78701132eafcc569223d504ece8b192ef8e1cc20
Fetched in submodule path 'platform/ios/vendor/mapbox-mobile-events', but it did not contain 78701132eafcc569223d504ece8b192ef8e1cc20. Direct fetching of that commit failed
```

**Comment by boundsj** [comment]
Sorry for the mixup @akitchen. Can you please give this another shot when you have a chance?

**Comment by boundsj** [comment]
Boba-fied cc @rclee @akitchen

**Comment by boundsj** [comment]
@rclee @akitchen @friedbunny https://github.com/mapbox/mapbox-gl-native/pull/10698/commits/d8e6312e165eb3a1da79c11f9749fb91f1b116a7 updates this PR to use [v0.3.0](https://github.com/mapbox/mapbox-events-ios/releases/tag/v0.3.0) of the events library instead of pinning to the previous, in-progress `mapbox-events-ios` branch.

I think this is ready for review.

File: platform/ios/src/MGLMapboxEvents.m [hunk 1]
```diff
@@ -1,643 +1,161 @@
...
-        // Configure logging
-        if ([self isProbablyAppStoreBuild]) {
-            self.canEnableDebugLogging = NO;
-
-            if ([[NSUserDefaults standardUserDefaults] boolForKey:@"MGLMapboxMetricsDebugLoggingEnabled"]) {
-                NSLog(@"Telemetry logging is only enabled in non-app store builds.");
-            }
-        } else {
-            self.canEnableDebugLogging = YES;
+        // Because we can get here [MGLAccountManaber +load] it is possible that values from user defaults will be read and
```
Comment (rclee) [inline platform/ios/src/MGLMapboxEvents.m]: typo:

> [MGLAccountManaber +load]

File: platform/ios/src/MGLMapboxEvents.m [hunk 2]
```diff
@@ -1,643 +1,161 @@
...
+        [self updateNonDisablingConfigurationValues];
+        [self updateDisablingConfigurationValuesWithNotification:notification];
+    });
+}
+
+- (void)updateNonDisablingConfigurationValues {
+    self.eventsManager.debugLoggingEnabled = [[NSUserDefaults standardUserDefaults] boolForKey:@"MGLMapboxMetricsDebugLoggingEnabled"];
+    
+    // It is possible that MGLTelemetryAccessToken is set and userDefaultsDidChange: is called before setupWithAccessToken: is called.
+    // In that case, setting the access token here will be a noop. In practice, that's fine because the access token value
```
Comment (rclee) [inline platform/ios/src/MGLMapboxEvents.m]: typo:

> setting the access token here will be a noop.

File: platform/ios/src/MGLMapboxEvents.m [hunk 3]
```diff
@@ -1,643 +1,161 @@
...
-    return _instanceID;
 }
 
-- (void)userDefaultsDidChange:(NSNotification *)notification {
-    
+- (void)updateDisablingConfigurationValuesWithNotification:(NSNotification *)notification {
     // Guard against over calling pause / resume if the values this implementation actually
-    // cares about have not changed
-    
+    // cares about have not changed. We guard becasue the pause and resume method checks CoreLocation's
```
Comment (rclee) [inline platform/ios/src/MGLMapboxEvents.m]: typo: 

> We guard becasue

File: platform/ios/src/MGLMapboxEvents.m [hunk 4]
```diff
@@ -1,643 +1,161 @@
...
+    if ([[[[NSUserDefaults standardUserDefaults] dictionaryRepresentation] allKeys] containsObject:MGLTelemetryAccessToken]) {
+        self.eventsManager.accessToken = [[NSUserDefaults standardUserDefaults] objectForKey:MGLTelemetryAccessToken];
     }
-    if (!_instanceID) {
-        _instanceID = [[NSUUID UUID] UUIDString];
-        NSTimeInterval twentyFourHourTimeInterval = 24 * 3600;
-        self.instanceIDRotationDate = [[NSDate date] dateByAddingTimeInterval:twentyFourHourTimeInterval];
+    
+    // It is possible that MGLTelemetryBaseURL is set and userDefaultsDidChange: is called before setupWithAccessToken: is called.
+    // In that case, setting baseURL here will be a noop. In practice, that's fine because the baseURL value
```
Comment (rclee) [inline platform/ios/src/MGLMapboxEvents.m]: typo:

setting the baseURL here will be a noop.

File: platform/ios/src/MGLMapboxEvents.m (hunk 2 above)
Comment (friedbunny) [reply platform/ios/src/MGLMapboxEvents.m]: 🐮 💬 noOOOOOOOOooooooooop

File: platform/ios/src/MGLMapboxEvents.m (hunk 1 above)
Comment (friedbunny) [reply platform/ios/src/MGLMapboxEvents.m]: This `+` would typically go outside the brackets, as well.

File: platform/ios/ios.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -1335,6 +1333,15 @@
 			path = "Integration Test Harness";
 			sourceTree = "<group>";
 		};
+		1F432781203B74DF00A21E6D /* Recovered References */ = {
```
Comment (friedbunny) [inline platform/ios/ios.xcodeproj/project.pbxproj]: The Recovered References group indicates that there’s a problem with the membership of the files it contains.

> When a project is opened, Xcode may create a Recovered References group in the Project navigator. This group contains files that are referenced in a target but aren't referenced in the group tree of the project. If the group appears, move each file to the proper location in the project as appropriate.

**Pull request merged**